uploaders/xcarchiveuploader.go

```diff
@@ -0,0 +1,90 @@
+package uploaders
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"path/filepath"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-xcode/plistutil"
+	"github.com/bitrise-io/go-xcode/xcarchive"
+)
+
+// DeployXcarchive ...
+func DeployXcarchive(pth, buildURL, token string) error {
+	log.Printf("analyzing xcarchive")
+	unzippedPth, err := xcarchive.UnzipXcarchive(pth)
+	if err != nil {
+		return err
+	}
+
+	ismacos, err := xcarchive.IsMacOS(filepath.Join(unzippedPth, pathutil.GetFileName(pth)))
+	if err != nil {
+		return fmt.Errorf("could not check if given project is macOS or not, error: %s", err)
+	}
+	if ismacos {
```

Comment:
Do not write a warning, just write a comment in the code here.

Reply:
Removed warning, added a comment to code.